ops/services/alerts/app_service_metrics/main.tf

```diff
@@ -85,12 +85,14 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "http_2xx_failed_requests
   query = <<-QUERY
 requests
 ${local.skip_on_weekends}
-| where toint(resultCode) between (200 .. 299) and success == false and timestamp >= ago(5m)
+| where toint(resultCode) between (200 .. 299) and timestamp >= ago(5m)
+| sort by timestamp asc
+| summarize failureRate = iff((todouble(sumif(1, success == false)) * 100 / todouble(sumif(1, success == true)) >= ${var.http_2xx_failure_rate_threshold}), 1, 0)
```

Comment:
This is cool! How does the trigger know to use failureRate and not the count, which seems to be the default based on the other queries in this file?

Reply:
Also, is this technically a ratio of failed:successful requests rather than a percentage of failed over total?

Reply:
Summarize returns a table that is the aggregate of the content in our input table. In our case, the input table is all requests within the last 5 minutes.

Our input content is filtered and evaluated on the right side of the `=` and assigned to failureRate. **(I renamed this to alert in a later push to hopefully clear up it's actual return.)**

- `iff()` is running an `if/else` clause against our expression contained within it.
- `todouble()` is turning our `integers` into `floats`
- `sumif()` is a filter and essentially acting as a `count()` 
  - `if request.success equals false add 1 to our failed count`
- The `1` and `0` on the end of the summarize line are our return values.
- If we take everything on the right side of our equal and use some numbers (let's say, `8 failed requests`, `16 total requests`, and an `acceptable failure rate of 5%`) it turns into this: `iff(8 * 100 / 16) >= 5, 1, 0)`, our comparison ends up being `50 >= 5` which would return `1` and trigger our alarm.